File: src/main/java/jp/kentan/minecraft/neko_event/config/ConfigManager.java
package jp.kentan.minecraft.neko_event.config;

import jp.kentan.minecraft.neko_event.config.provider.*;
import jp.kentan.minecraft.neko_event.util.Log;

import java.io.*;

public class ConfigManager {

    private final File mDataFolder;

    public ConfigManager(File dataFolder){
        mDataFolder = dataFolder;

        SignConfigProvider.setup(dataFolder);
        GachaConfigProvider.setup(dataFolder);
        DungeonConfigProvider.setup(dataFolder);
        ParkourConfigProvider.setup(dataFolder);
        PlayerConfigProvider.setup(dataFolder);
        KeyConfigProvider.setup(dataFolder);
        PasswordConfigProvider.setup(dataFolder);
    }

    public static void createIfNeed(File file){
        if(!file.exists()){
            try {
                if(!file.createNewFile()){
                    Log.warn(file.getName() + "の作成に失敗しました.");
                }
            } catch (IOException e) {
                e.printStackTrace();
                Log.error(e.getMessage());
            }
        }
    }

    public void reload(){
        SignConfigProvider.setup(mDataFolder);
        PlayerConfigProvider.setup(mDataFolder);
        GachaConfigProvider.load();
        DungeonConfigProvider.load();
        ParkourConfigProvider.load();
        KeyConfigProvider.load();
        PasswordConfigProvider.load();
    }
}

File: src/main/java/jp/kentan/minecraft/nekoevent/config/ConfigManager.kt
package jp.kentan.minecraft.nekoevent.config

import jp.kentan.minecraft.nekoevent.config.provider.*
import java.io.File
import java.text.SimpleDateFormat

class ConfigManager(dataFolder: File) {

    companion object {
        val DATE_FORMAT = SimpleDateFormat("yyyy/MM/dd")
    }

    private val playerConfigProvider = PlayerConfigProvider(dataFolder)

    val signConfigProvider = SignConfigProvider(dataFolder)
    val gachaConfigProvider = GachaConfigProvider(dataFolder)
    val keyConfigProvider = KeyConfigProvider(dataFolder)
    val passwordConfigProvider = PasswordConfigProvider(dataFolder)
    val parkourConfigProvider = ParkourConfigProvider(dataFolder, playerConfigProvider)
    val dungeonConfigProvider = DungeonConfigProvider(dataFolder, playerConfigProvider)
    val ticketConfigProvider = TicketConfigProvider(dataFolder, playerConfigProvider)
    val bonusConfigProvider = BonusConfigProvider(dataFolder, playerConfigProvider)

    fun load() {
        gachaConfigProvider.load()
        keyConfigProvider.load()
        passwordConfigProvider.load()
        parkourConfigProvider.load()
        dungeonConfigProvider.load()
        ticketConfigProvider.load()
        bonusConfigProvider.load()
    }

    fun reload() {
        signConfigProvider.load()
        load()
    }
}